api/renewer.go

```diff
@@ -307,3 +338,23 @@ func (r *Renewer) sleepDuration(base time.Duration) time.Duration {
 
 	return time.Duration(sleep)
 }
+
+// calculateGrace calculates the grace period based on a reasonable set of
+// assumptions given the total lease time; it also adds some jitter to not have
+// clients be in sync. We calculate this continuously so long as the new lease
```

Comment:
```
We calculate this continuously so long as the new lease
 +// duration is greater than the previous; no change means we don't need to
 +// recalculate, and if the lease duration keeps decreasing we've hit max and
 +// want to be able to rely on this.
```

This is actually done by the callers so shouldn't be on this method.

Reply:
Removed that part of the comment.